demo/midgard-node/src/database/utils/user-events.ts

```diff
@@ -124,3 +136,90 @@ export const delEntries = (
       Columns.ID,
     )} IN ${sql.in(ids)}`;
   }).pipe(sqlErrorToDatabaseError(tableName, "Failed to delete given UTxOs"));
+
+export class UserEventsConversionError extends EffectData.TaggedError(
+  "UserEventsConversionError",
+)<SDK.GenericErrorFields> {}
+
+export const makeTransactionUnspentOutput = (
+  entry: Entry,
+  policyId: PolicyId,
+): Effect.Effect<
+  CML.TransactionUnspentOutput,
+  UserEventsConversionError,
+  never
+> =>
+  Effect.gen(function* () {
+    const l1Utxo = CML.TransactionUnspentOutput.from_cbor_bytes(
+      Buffer.from(entry[Columns.L1_UTXO_CBOR]),
+    );
+    const policyIdScriptHash = CML.ScriptHash.from_hex(policyId);
```

Comment:
This is an operation that can fail. We should use `Effect.try`:
```suggestion
    const policyIdScriptHash = Effect.try({
      try: () => CML.ScriptHash.from_hex(policyId),
      catch: (e) => ...,
    });
```